universe/node/api_space_attributes.go

```diff
@@ -110,3 +111,67 @@ func (n *Node) apiGetSpaceSubAttribute(c *gin.Context) {
 
 	c.JSON(http.StatusOK, value)
 }
+
+func (n *Node) apiSetSpaceSubAttribute(c *gin.Context) {
+	inBody := struct {
+		PluginID          string `json:"plugin_id" binding:"required"`
+		AttributeName     string `json:"attribute_name" binding:"required"`
+		SubAttributeKey   string `json:"sub_attribute_key" binding:"required"`
+		SubAttributeValue string `json:"sub_attribute_value" binding:"required"`
+	}{}
+
+	if err := c.ShouldBindJSON(&inBody); err != nil {
+		err = errors.WithMessage(err, "Node: apiSetSpaceSubAttribute: failed to bind json")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_body", err, n.log)
+		return
+	}
+
+	spaceID, err := uuid.Parse(c.Param("spaceID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiSetSpaceSubAttribute: failed to parse space id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_space_id", err, n.log)
+		return
+	}
+
+	pluginID, err := uuid.Parse(inBody.PluginID)
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiSetSpaceSubAttribute: failed to parse plugin id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_plugin_id", err, n.log)
+		return
+	}
+
+	space, ok := n.GetSpaceFromAllSpaces(spaceID)
+	if !ok {
+		err := errors.Errorf("Node: apiSetSpaceSubAttribute: space not found: %s", spaceID)
+		api.AbortRequest(c, http.StatusNotFound, "space_not_found", err, n.log)
+		return
+	}
+
+	attributeTypeID := entry.NewAttributeTypeID(pluginID, inBody.AttributeName)
+	_, ok = n.attributeTypes.GetAttributeType(attributeTypeID)
+	if !ok {
+		err := errors.Errorf("Node: apiSetSpaceSubAttribute: attribute type not found: %s", inBody.AttributeName)
+		api.AbortRequest(c, http.StatusNotFound, "attribute_type_not_found", err, n.log)
+		return
+	}
+
+	attributeID := entry.NewAttributeID(pluginID, inBody.AttributeName)
+	spaceAttributePayload, ok := space.GetSpaceAttributePayload(attributeID)
+	if !ok {
+		err := errors.Errorf("Node: apiSetSpaceSubAttribute: space attribute payload not found: %s", attributeID)
+		api.AbortRequest(c, http.StatusNotFound, "space_attribute_payload_not_found", err, n.log)
+		return
+	}
+
+	(*spaceAttributePayload.Value)[inBody.SubAttributeKey] = inBody.SubAttributeValue
+
+	if err := space.UpsertSpaceAttribute(attributeID, modify.ReplaceWith(spaceAttributePayload), true); err != nil {
+		err = errors.WithMessage(err, "Node: apiSetSpaceSubAttribute: failed to upsert space attribute")
+		api.AbortRequest(c, http.StatusInternalServerError, "failed_to_upsert", err, n.log)
+		return
+	}
```

Comment:
it's wrong, in this case you will replace someone else's changes
as I mentioned in the chat you need to write your own local modify function

you can check this example: https://github.com/momentum-xyz/ubercontroller/blob/develop/universe/node/api_users.go#L235

the case above is more difficult then yours